source/develop/release-management/features/network/autodefine-external-network.md

```diff
@@ -0,0 +1,134 @@
+manage their network via an SDN (such as OVN or Neutron) yet have near-native 
+performance. However, without auto-define feature, such admin would have to 
+create a physical network (and attach it to hosts), create an external network,
+and remember to connect this two networks together. Autodefine helps reducing 
+these administrative tasks. In an OvS cluster with an external provider 
+attached, the external network is automatically created and connected to the 
+physical one.
+
+## Prerequisites
+
+Cluster has to be configured with switch type OVS and valid external provider.
+The user needs to make sure that connection to external provider is valid.
+
+### Limitations
+
+* With OVS switch type in cluster, only external networks are allowed to 
+  be attached to the VM interface. 
+
+* Physical network linked to any external network cannot be removed. This is for
+  keeping the consistency of database. In the future we might add a way to 
+  remove all dependent external networks with physical one. 
+
+* The switch type of existing clusters cannot be changed (from Linux bridge to 
+  OVS or vice versa). Thus, we currently do not need to worry about 
+  auto-defining external networks on top of existing physical networks.
+
+* Currently updating the physical network won't update auto-defined external 
+  network. This might change in the future. 
+
+* VMs will be scheduled only on hosts that does have attached linked physical 
+  network to one of it's interfaces. Same role applies for hotplug of VM 
+  interfaces. 
+
+## Detailed description
+
+### Creating of new physical network
+
+When creating new network choose its name. Check the `VM network`. If vlan 
+tagging is needed, check the `Enable VLAN tagging` and put the tag into the 
+field. Please note that auto-define can be triggered only on top of network that
+is not marked as external. 
+
+![Creation of physical network](../../../../images/wiki/auto-define_create.png  "New physical network")
+
+After confirmation if all conditions were met, the external network connected to 
+the physical one, will be created simultaneously. The auto-defined external 
+network is named `external_name_of_physical_network` (The physical network name
+might be shortened so that auto-defined network name is not bigger than 256 
+characters). 
+
+![Auto-define network created](../../../../images/wiki/auto-define_created.png  "Auto-define external network")
+
+In case there are clusters with different default external providers in the data 
+center the Engine will create multiple external networks for every provider and 
+add a number behind to this auto-defined network name. 
+
+Please note that all hosts running VMs connected to the external network has
+to be attached to the physical network. 
+
+
+
+## Implementation
+
+### Engine
+
+* A new query `GetAllVmNetworksByClusterId` that returns only VM networks is 
+  added.For Cluster with OVS switch type, this query returns only external 
```

Comment:
Space is missing between sentences. 'For a cluster with OVS switch type'

Reply:
Done